src/Microsoft.CodeAnalysis.ResxSourceGenerator/Microsoft.CodeAnalysis.ResxSourceGenerator/AbstractResxGenerator.cs

```diff
@@ -22,93 +23,147 @@
+                        var errorText = SourceText.From(text, Encoding.UTF8, SourceHashAlgorithm.Sha256);
+                        context.AddSource($"{Path.GetFileName(resourceInformation.ResourceFile.Path)}.Error", errorText);
+                    }
+                });
+        }
 
-            var impl = new Impl(
-                this,
-                language: context.Compilation.Language,
-                resourceFile: resourceFile,
-                resourceName: string.Join(".", rootNamespace, resourceName),
-                resourceClassName: null,
-                omitGetResourceString: omitGetResourceString,
-                asConstants: asConstants,
-                includeDefaultValues: includeDefaultValues,
-                emitFormatMethods: emitFormatMethods);
-            if (impl.Execute(context))
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="AssemblyName"></param>
+        /// <param name="CodeLanguage">Language of source file to generate. Supported languages: CSharp, VisualBasic.</param>
+        /// <param name="SupportsNullable"></param>
+        private readonly record struct CompilationInformation(
+            string? AssemblyName,
+            string CodeLanguage,
+            bool SupportsNullable,
+            bool HasAggressiveInlining,
+            bool HasNotNullIfNotNull);
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="CompilationInformation">Information about the compilation.</param>
+        /// <param name="ResourceFile">Resources (resx) file.</param>
+        /// <param name="ResourceName">Name of the embedded resources to generate accessor class for.</param>
+        /// <param name="ResourceClassName">Optionally, a <c>namespace.type</c> name for the generated Resources accessor class. Defaults to <see cref="ResourceName"/> if unspecified.</param>
+        /// <param name="OmitGetResourceString">If set to <see langword="true"/>, the <c>GetResourceString</c> method is not included in the generated class and must be specified in a separate source file.</param>
+        /// <param name="AsConstants">If set to <see langword="true"/>, emits constant key strings instead of properties that retrieve values.</param>
+        /// <param name="IncludeDefaultValues">If set to <see langword="true"/>, calls to <c>GetResourceString</c> receive a default resource string value.</param>
+        /// <param name="EmitFormatMethods">If set to <see langword="true"/>, the generated code will include <c>.FormatXYZ(...)</c> methods.</param>
+        private readonly record struct ResourceInformation(
+            CompilationInformation CompilationInformation,
+            AdditionalText ResourceFile,
+            string ResourceName,
+            string? ResourceClassName,
+            bool OmitGetResourceString,
+            bool AsConstants,
+            bool IncludeDefaultValues,
+            bool EmitFormatMethods);
```

Comment:
This struct looks quite large. Not sure if it can have perf impact.

https://learn.microsoft.com/en-us/dotnet/standard/design-guidelines/choosing-between-class-and-struct recommends structs with size less than 16 bytes

This one is 23 bytes if I'm calculating it correctly.

Reply:
➡️ Converted both to sealed classes

Reply:
> recommends structs with size less than 16 bytes

Sounds like that could be a helpful analyzer in and of itself.